src/crimson/os/seastore/seastore_types.h

```diff
@@ -1088,8 +1088,8 @@ std::ostream &operator<<(std::ostream &out, extent_types_t t);
 using reclaim_gen_t = uint8_t;
 
 constexpr reclaim_gen_t DIRTY_GENERATION = 1;
-constexpr reclaim_gen_t COLD_GENERATION = 1;
-constexpr reclaim_gen_t RECLAIM_GENERATIONS = 3;
+constexpr reclaim_gen_t COLD_GENERATION = 2;
+constexpr reclaim_gen_t RECLAIM_GENERATIONS = 4;
```

Comment:
> OK, does it look better?
> 
> ```c++
> reclaim_gen_t target_gen = generation + 1;
> if (target_gen < COLD_GENERATION) {
>   target_gen = COLD_GENERATION;
> } else if (target_gen > RECLAIM_GENERATIONS) {
>   target_gen = RECLAIM_GENERATIONS;
> }
> ```

RECLAIM_GENERATIONS should be 3, and I'll update all `assert(gen < RECLAIM_GENERATIONS)` to `assert(gen <= RECLAIM_GENERATIONS)`

Reply:
Let's instead rename `RECLAIM_GENERATIONS` to `REWRITE_GENERATIONS`, and `reclaim_gen_t` to `rewrite_gen_t`, and keep the original asserts.

"rewrite" will be more precise than "reclaim" as the goal of "generation" is to track the number of rewrites.

`COLD_GENERATION` might be more appropriate as `MIN_COLD_GENERATION`.